plugins/Validation/src/Validation.ts

```diff
@@ -1,223 +1,346 @@
+/* tslint:disable:no-any */
+import { State } from '@hookstate/core';
+import { PluginCallbacks, StateValueAtRoot } from '@hookstate/core/dist';
 
-import { Plugin, Path, StateValueAtPath, State } from '@hookstate/core';
+export const ValidationId = Symbol('Validation');
 
-export type ValidationSeverity = 'error' | 'warning';
+type ValidateFn<T> = (value: T) => boolean;
+type Path = readonly (string | number | symbol)[];
```

Comment:
Hookstate does not deal with symbol properties intentionally. Why do you need to redefine Path from Hookstate? I see there are public functions with Path as an argument....

Reply:
i've been developing this plugin in my project and copying order, so didn't optimize yet